automation/check-patch.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash -xe
+# Patch/PR testing hook script
+#
+
+main() {
+    # We need to keep the VMs between runs for now because the containers are
```

Comment:
We currently always do a `vagrant halt` after the tests. This way, the VMs don't need to run, but you can just do `vagrant up` on the next CI run. If you do `vagrant destroy`, the VM needs to be re-provisioned. To make that faster too, you can use the VAGRANT_CACHE_DOCKER and VAGRANT_CACHE_RPM options.

Reply:
Actually this comment was wishful thinking. My environment thoroughly destroys libvirt objects between runs, so nothing is left when the job is done, regardless of what I put here.

(Incidentally some of that code was written by @fabiand ;)